enterprise/server/remote_execution/cgroup/cgroup.go

```diff
@@ -0,0 +1,189 @@
+package cgroup
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/buildbuddy-io/buildbuddy/server/util/log"
+	"github.com/buildbuddy-io/buildbuddy/server/util/status"
+
+	repb "github.com/buildbuddy-io/buildbuddy/proto/remote_execution"
+)
+
+const (
+	// Standard path where cgroupfs is expected to be mounted.
+	cgroupfsPath = "/sys/fs/cgroup"
+
+	// Placeholder value representing the container ID in cgroup path templates.
+	cidPlaceholder = "{{.ContainerID}}"
+)
+
+// Paths holds cgroup path templates that map a container ID to their cgroupfs
+// file paths.
+//
+// A single instance should be shared across all containers of the same
+// isolation type, since the first call to Stats() walks the cgroupfs tree in
+// order to discover the cgroup path locations.
+type Paths struct {
+	mu sync.RWMutex
+
+	// MemoryTemplate is the path template for the cgroupfs file containing
+	// current memory usage.
+	MemoryTemplate string
+	// CPUPathTemplate is the path template for the cgroupsfs file containing
+	// cumulative CPU usage.
+	CPUTemplate string
+}
+
+func (p *Paths) Stats(ctx context.Context, cid string) (*repb.UsageStats, error) {
+	if err := p.find(ctx, cid); err != nil {
+		return nil, err
+	}
+
+	memUsagePath := strings.ReplaceAll(p.MemoryTemplate, cidPlaceholder, cid)
+	cpuUsagePath := strings.ReplaceAll(p.CPUTemplate, cidPlaceholder, cid)
+
+	memUsageBytes, err := readInt64FromFile(memUsagePath)
+	if err != nil {
+		return nil, err
+	}
+	var cpuNanos int64
+	if p.CgroupVersion() == 1 {
```

Comment:
guard against 0 here?

Reply:
find() returns an error in that case, so I think it's not needed.

Reply:
lol i agree with @iain-macdonald 

i have a small brain and seeing:

```
if 1 {
} else if 2 {
} else {
 raise error(get out)
}
```

makes me feel safe

Reply:
okay, fair :) done